docs/lang/articles/contribution/contributor_guide.md

```diff
@@ -131,144 +125,167 @@ python misc/code_format.py
-  - A PR should ideally address only one issue.
-    - It is fine to include off-topic **trivial** refactoring such as typo fixes;
-    - The reviewer is qualified to request you to remove off-topic **non-trivial** changes in your PR.
-  - When implementing a complex feature, consider breaking it down into small PRs to keep a more detailed development history and to interact with core developers more frequently.
-  - PR titles should be short sentences describing the changes and following [certain format](./contributor_guide#pr-title-format-and-tags).
-  - In the description of a PR, it will be nice to link relevant GitHub issues (e.g. `fixes #issue_number`) or provide a little context on the motivation.
+- **When describing your PR:**
+  - Provide sufficient information in the description of your PR to provide the community with clearer context:
+    - Link to a specific GitHub issue if applicable, for example `fixes #<issue_number>`.
+    - Share important design decisions in your description.
 
-Some important implementation decisions you made in the PR is also helpful.
+- **If you create a PR still in progress:**
 
-- If you want early feedback from core developers,
-  - Open a PR in [Draft](https://github.blog/2019-02-14-introducing-draft-pull-requests/) state on GitHub to share your progress;
-  - Ensure that you @ the corresponding developers in your comments or request reviews from them.
-- All PRs should ideally come with corresponding **tests**. See [Testing](./contributor_guide#testing).
-- Most PRs, except for those for internal compiler implementations, ought to be accompanied with **document updates**. See [Documentation](./contributor_guide#documentation).
-- All PRs must pass **continuous integration tests** before they get merged. See [Using continuous integration](./contributor_guide#using-continuous-integration).
-- All PRs must pass **code format checks**. See [Enforcing code style](./contributor_guide#enforcing-code-style).
-- Read a great article from Google on [how to have your PR merged quickly](https://testing.googleblog.com/2017/06/code-health-too-many-comments-on-your.html)
+  - Click **Convert to draft** on your PR page to convert the PR to draft, indicating that you are still working on it.
+  - Click **Ready for review** when you are all set and up for a review.
+  - See the [Draft](https://github.blog/2019-02-14-introducing-draft-pull-requests/) for more information.
 
-[[PDF]](https://github.com/yuanming-hu/public_files/blob/master/graphics/taichi/google_review_comments.pdf)
+- **If you make multiple commits to an open PR:**
 
-- PR authors are required to **squash commits on your own**.
-- If you are making multiple PRs,
-  - Independent PRs should be based on **different** branches forking from `master`;
-  - PRs with dependencies should be raised only after all prerequisite PRs are merged into `master`.
+  - Consider rebasing your commits to keep the commit history clean and meaningful.
```

Comment:
I feel like in practice we don't really follow this rule. Since we always squash commits in a PR into a single commit with the final commit message being the PR title in the `master` branch, the intermediate commit messages/history are unimportant. Maybe we can remove this bullet?